### PR TITLE
Added user roles to the single and all user(s) getters

### PR DESCRIPTION
**Pull request by ddelpiano**

### CODE REVIEW

**Comment by zsinnema** [comment]
please also add these functions to make the implementation of the realm roles complete:
```
get_user_realm_roles
get_current_user_realm_roles
user_has_realm_role
current_user_has_realm_role
```

**Review by zsinnema — CHANGES_REQUESTED** [review]
see my comments

File: libraries/cloudharness-common/cloudharness/auth/keycloak/__init__.py [hunk 1]
```diff
@@ -330,6 +358,17 @@ def user_has_client_role(self, user_id, client_name, role):
         roles = [user_client_role for user_client_role in self.get_user_client_roles(user_id, client_name) if user_client_role['name'] == role]
         return roles != []
 
+    def user_has_realm_role(self, user_id, role):
+        """
+        Tests if the user has the given role within the given client
```
Comment (zsinnema) [inline libraries/cloudharness-common/cloudharness/auth/keycloak/__init__.py:363]: "within the current realm"

File: libraries/cloudharness-common/cloudharness/auth/keycloak/__init__.py [hunk 2]
```diff
@@ -343,6 +382,17 @@ def current_user_has_client_role(self, client_name, role):
             client_name,
             role)
 
+    def current_user_has_realm_role(self, role):
+        """
+        Tests if the current user has the given role within the given client
```
Comment (zsinnema) [inline libraries/cloudharness-common/cloudharness/auth/keycloak/__init__.py:387]: "within the current realm"

**Review by zsinnema — CHANGES_REQUESTED** [review]

**Comment by ddelpiano** [comment]
@zsinnema I am merging this to fix the ifn deployment since currently we don't get clinician due to the logic to check the clinician or reviewer roles.
I fixed the latest changes requested already but if there is anything that you like me to change let me know and I will open a separate PR. Thx

**Pull request merged**